benchmark-service/benchmark/benchmark.py

```diff
@@ -57,6 +59,58 @@ def get_benchmarks(file_path):
     return benchmarks
 
 
+def get_comparisons(benchmarks1, benchmarks2, metric):
+    def get_metric(data):
+        if metric == 'median':
+            return data['median']
+        assert metric == 'best'
+        return min(data['times'])
+
+    comparisons = []
+    for d in benchmarks1['data']:
```

Comment:
So this is one way to implement an outer join, but it's not very efficient because you are doing O(n^2) work. There's three ways to do this efficiently. One is sort the keys for each dictionary and then do a zip join. The second is to make a set of all the keys, sort them, and then try to get the key from each dictionary. The third way is to iterate over all the keys in dict1. Then iterate over the keys in dict2 and only output data if it isn't present in dict1.

I think in this case getting a set of all the possible names and then iterating through them is the best way to go.